src/syrupy/data.py

```diff
@@ -50,6 +50,8 @@ def get(self, snapshot_name: str) -> Optional["Snapshot"]:
 
     def add(self, snapshot: "Snapshot") -> None:
         self._snapshots[snapshot.name] = snapshot
+        if snapshot.name != SNAPSHOT_EMPTY_FOSSIL_KEY:
+            self.remove(SNAPSHOT_EMPTY_FOSSIL_KEY)
```

Comment:
What's the broken behaviour this specific change is fixing?

Reply:
Nothing broken at the moment, but there should not be an empty snapshot fossil marker once snapshot have been added. Noticed it while attempting to fix the listed issue.